### ISSUE TITLE
Add "gap_lag" and "heterogeneous time aggregation" (RGDR relevant) features/calendar

### ISSUE THREAD

**Issue by geek-yang** [AI4S2S/s2spy#47]
Two new features relevant for RGDR need to be accommodated by the `time.py` module later:

(1) Lag as gap 
For RGDR, we might want to define lags in a more precise manner. This gives more flexibility to define the lead-time 
of the forecast. 

![image](https://user-images.githubusercontent.com/25911757/179239012-5d2ad99a-0f5f-4f5f-a09a-567807645262.png)

(2) Heterogeneous time aggregation
The further you go back in time, the more you might want to focus on low-frequency variability.

![image](https://user-images.githubusercontent.com/25911757/179239419-8f4e7fbb-3e14-4c2f-8d41-ac0055400c59.png)

Whether these two features need to be added as features or new calendar types need to be discussed. But for now, this task is not prioritized. This issue ensures that the request is well documented.

This issue is also relevant to issue #30.

**Comment by Peter9192** [comment]
Just to articulate: with lag_as_gap you would only need the single interval that corresponds to that lag, right? Not the 10 lags before that. 

By definition, the period has a start/center/end. So if I understand you correctly, you want to precisely align the **end** of the precursor to the **start** of the target (which is currently not possible, since you are limited to lags ending on days that are exactly offset n days from the end of the target period, where n is the number of days, i.e. the inverse frequency). Right?

**Comment by semvijverberg** [comment]
Yes that is correct @Peter9192. As a heads up, this feature will be needed for the upcoming Lorentz workshop. I will share more the details on that soon.
